### DISCUSSION TITLE
updating 0.57.3 packages through opkg broke shadowsocks?

### DISCUSSION THREAD

**Discussion by bdruth** [Ysurac/openmptcprouter#1801] [Q&A]
Hi,

I may have done something really stupid. I wanted to see if there were any updates to install, since it had been a while since I setup (few months) and I wasn't ready to try 0.58 RC yet. 
```
> opkg update
> opkg list-upgradable
> opkg list-upgradable | cut -f 1 -d ' ' | xargs opkg upgrade
```
It was mostly luci updates, but it looks like some shadowsocks packages were updated from `http://download.openmptcprouter.com:80/release/v0.57.3/rpi4/...` -- now I get this repeating error in the logs and my shadowsocks won't connect:
```
Sat May  1 23:16:14 2021 user.info : using tcp fast open
Sat May  1 23:16:14 2021 daemon.info /usr/bin/ss-redir[10021]: enable TCP no-delay
Sat May  1 23:16:14 2021 daemon.info /usr/bin/ss-redir[10021]: resolving hostname to IPv6 address first
Sat May  1 23:16:14 2021 daemon.info /usr/bin/ss-redir[10021]: initializing ciphers... chacha20
Sat May  1 23:16:14 2021 daemon.info /usr/bin/ss-redir[10021]: Stream ciphers are insecure, therefore deprecated, and should be almost always avoided.
Sat May  1 23:16:14 2021 daemon.err /usr/bin/ss-redir[10021]: Invalid key for your chosen cipher!
Sat May  1 23:16:14 2021 daemon.err /usr/bin/ss-redir[10021]: It requires a 32-byte key encoded with URL-safe Base64
Sat May  1 23:16:14 2021 daemon.err /usr/bin/ss-redir[10021]: Generating a new random key: X_qDd7hm2qNcp_Qk-HahhDSYXArA7UcIXue3gY4cAMU=
Sat May  1 23:16:14 2021 daemon.err /usr/bin/ss-redir[10021]: Please use the key above or input a valid key
```
I've tried a few different things ... it seems when I "Save & Apply" from the wizard, the "remote servers" tab in Shadowsocks-libev shows `chacha20-ietf-poly1305`, when the VPS config that was created when I installed says just `chacha20` ... but even if I switch it & use the password from the VPS config, it doesn't work and doesn't change the log output.

Help?

**Comment by Ysurac** [comment]
Yes, not possible to update from repo on <0.58. OpenMPTCProuter use a patched release of Shadowsocks-libev and release in repo is from OpenWrt and not patched. And it's the same for some others packages.
You need to install/update image via interface.

**Reply by bdruth** [reply]
How?

**Reply by Ysurac** [reply]
As I said, update image via the interface. In system->Flash firmware or as you did at install time.

**Reply by bdruth** [reply]
Oh, gotcha ... thank you.

**Reply by bdruth** [reply]
Thanks @Ysurac - was able to upload the 0.57.3 sysupgrade and get everything working again.
